pages/css-grid-superposition--qcj5jo/css-grid-superposition--qcj5jo.md

```diff
@@ -0,0 +1,279 @@
+* Always have the CSS-Tricks [complete guide to Grid](https://css-tricks.com/snippets/css/complete-guide-grid/) open in a tab.
+* Practice, practice, practice...
+
+Now that you're more familiar with CSS grid, we can get to the "anti layout shift superposition technique".
+Let's take the initial simple template we had earlier:
+
+Let's take a simplified version of the template:
+
+```html
+<div class="chart-component">
+  <div class="title">
+    HTTP response codes <button>toggle chart/info</button>
+  </div>
+  <div class="chart">
+    <!-- chart here -->
+  </div>
+  <div class="info">
+    <!-- short text here -->
+  </div>
+</div>
+```
+
+If we use CSS grid on `.chart-component` like this:
+
+```css
+.chart-component {
+  display: grid;
+  gap: 1rem;
+}
+```
+
+Our component would look like this:
+
+![UI Component displaying some help](./component-grid-all.png)
+
+Thanks to the awesome [CSS Grid Inspector](https://developer.mozilla.org/en-US/docs/Tools/Page_Inspector/How_to/Examine_grid_layouts) from Firefox's DevTools, we have annotations for columns and rows.
+We can see that:
+
+* `.title`, `.chart` and `.info` are between column 1 and 2
+* `.title` is between row 1 and 2
+* `.chart` is between row 2 and 3
+* `.info` is between row 3 and 4
+
+This is the default behaviour in a simple 1 column grid, children are placed in order.
+With `grid-column` and `grid-row`, we can force an element to be placed in a specific area of the grid.
+
+For example, if we place our two elements in the same area like this:
+
+```css
+.chart,
+.info {
+  grid-column: 1 / 2;
+  grid-row: 2 / 3;
+}
+```
+
+The result would look like this:
+
+![UI Component displaying some help](./component-grid-superposed.png)
+
+Looks weird and unreadable, right?
+Yes but remember, we'll hide the appropriate panel with a `visibility: hidden`.
+With this technique, we actually told the CSS engine to prepare a grid where the area at column `1 / 2` and row `2 / 3` should adapt to what's inside.
+
+In other words: this very area will always be as big as the biggest between `.chart` and `.info`.
+The height of the whole component will be the same when toggling between states.
+
+😎 No more layout shift!
+
+This solution is even better than the one with position absolute because if the text is really short, the component will adapt to the size of the chart.
+
+## What else?
+&
+We saw an example on the height of a component but you can also use this technique with width.
+
+If you look carefully at top right corner of the first screenshot, you'll see buttons.
+The "Stop app" button has bigger horizontal paddings than the other buttons.
+This is because, when you click on this very button, we replace the text with "Click to cancel" for 3 seconds so you can change your mind.
```

Comment:
I suggest: "This is because, when clicking this very button, the text is replaced with "Click to cancel" for 3 seconds in case you changed your mind."